core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -226,7 +227,8 @@ public RemoteLogManager(RemoteLogManagerConfig rlmConfig,
         rlmCopyQuotaManager = createRLMCopyQuotaManager();
         rlmFetchQuotaManager = createRLMFetchQuotaManager();
 
-        indexCache = new RemoteIndexCache(rlmConfig.remoteLogIndexFileCacheTotalSizeBytes(), remoteLogStorageManager, logDir);
+        RemoteLogManagerConfig rlmConfig = config.remoteLogManagerConfig();
+        indexCache = new RemoteIndexCache(config.remoteLogIndexFileCacheTotalSizeBytes(), remoteLogStorageManager, logDir);
```

Comment:
not sure whether I have caught the context, so please feel free to correct me.

1. IIRC, the dynamical configs are loaded by another thread, and hence we may NOT see the latest configs, which were updated dynamically, in creating `RemoteLogManager`, right? 
2. those configs (`remote.log.manager.copy.max.bytes.per.second`, `remote.log.manager.fetch.max.bytes.per.second`) can be updated by `reconfigure` process, so it should be fine to initialize them with "stale" (static) configs after broker restart, right?

Reply:
> 1. IIRC, the dynamical configs are loaded by another thread, and hence we may NOT see the latest configs, which were updated dynamically, in creating RemoteLogManager, right?

No, KafkaServer/BrokerServer does `config.dynamicConfig.initialize` before creating the RemoteLogManager instance so the dynamic configs gets updated in the `KafkaConfig` object but not in the `KafkaConfig.remoteLogManagerConfig()`. 

I have tested the patch only with ZooKeeper. I think the behavior should be similar of KRaftMetadataCache/ConfigRepository.

Reply:
> I have tested the patch only with ZooKeeper. I think the behavior should be similar for KRaftMetadataCache/ConfigRepository.

That is a good point, and maybe they do have something difference. 

https://github.com/apache/kafka/blob/bcf781230e750fd5efbf276e984f2875bf9fa683/core/src/main/scala/kafka/server/DynamicBrokerConfig.scala#L228

in kraft, `zkClientOpt` is none so it does not update it with dynamical parts.

Reply:
Yes, I confirmed in KRaft, it won't have this issue.

Reply:
> Yes, I confirmed in KRaft, it won't have this issue.

Sorry that I'm not sure which issue you confirmed. If we are taking about dynamic configs in starting. According to above comments, it seems to me this fix which tries to return latest (dynamic) configs works well only if kafka is in zk. In kraft, this fix is no-op as it still return static configs. 

Please correct me If I'm lost

Reply:
@chia7712 , yes, you're right! To KRaft, this fix is no-op.